### ISSUE TITLE
bitrate is always 44.1khz regardless of setting

### ISSUE THREAD

**Issue by robfrye** [balbuze/volumio-plugins#258]
Playing via this plugin on a Raspberry Pi 4b - No matter how I set the settings, the bitrate is always 44.1khz 16bit.  Connecting Spotify directly using the Spotify plugin yields 320khz 16bit (and significantly better sound quality)

<img width="1273" alt="Screen Shot 2020-03-28 at 1 34 16 PM" src="https://user-images.githubusercontent.com/2500851/77833224-f605d880-70f8-11ea-9a8d-3750b18f41d8.png">

**Comment by illtrick** [comment]
Bitrate (320kbs), Bitdepth (16bit) and Samplerate (44.1khz) are all unique values.

**Comment by robfrye** [comment]
Fair enough, updated the question. Any idea why sound quality is so much worse using Connect than using the regular Spotify plugin?

**Comment by illtrick** [comment]
Interesting, can you describe the difference? Volume level is certainly different for me when switching back and forth but I don’t notice any sonic difference after volume matching.

**Comment by robfrye** [comment]
sound is very muted and lacks dynamic range. Not much volume difference for me.
If it's helpful, I am running USB out to an external DAC and then into a pretty decent amp/speaker setup.

**Comment by illtrick** [comment]
volume normalization could be the culprit, that squashes dynamic range.  is it set to on in either the plugin or globally?

**Comment by robfrye** [comment]
Off in both places.  Happy to send you logs or whatever if you can tell me how to do it.

**Comment by illtrick** [comment]
I can help you part of the way, I'm no dev but I can at least help you get logs.  I actually came here to post my own issue but got distracted by yours.  Best of luck!

1. Make sure you have debug mode enabled in the plugin
2. SSH to your device (if you don't know how to do that there's links in the instructions on the main page )
3. copy paste and this command:
4. `sudo journalctl -f -o cat -u volspotconnect2.service`
5. It should print events as they occur.  I'd go through a standard startup process (spotify app closed, open app, connect to device, play track) and copy paste them here.

**Comment by ashthespy** [comment]
>Interesting, can you describe the difference? Volume level is certainly different for me when switching back and forth but I don’t notice any sonic difference after volume matching.

That shouldn't be the case - can you describe your setup?

@robfrye Try turning off the volume control as well, if your using a software mixer.

**Comment by robfrye** [comment]
I wasn't using the volume control, however now it won't even connect so I can't help more at the moment.  I'll pile on to the Cannot Connect thread with my logs

**Comment by illtrick** [comment]
> > Interesting, can you describe the difference? Volume level is certainly different for me when switching back and forth but I don’t notice any sonic difference after volume matching.
> 
> That shouldn't be the case - can you describe your setup?
> 
@ashthespy I’ll take another go at it tomorrow. You’re making me think I’m crazy now.  I swear the volume levels were different...

**Comment by ashthespy** [comment]
@illtrick 
The volume level depends on the setup. For a hardware mixer,  `spop` - the dameon used in the spotify plugin uses `sox` and 'alsa' for the output. Thus the volume level is as defined by the "system volume" set by Volumio through `alsa` (some esoteric cases use a septate I2C device, but most is done with alsa).
The daemon we use, `vollibrespot` by comparison, ties directly to `alsa` for both the output and hardware mixer. Thus, changing volume from your client, should result in a direct change of the "system" volume for Volumio. Software mixers on the other hand, map the 0-100 of the client volume, to the current "system volume" of Volumio.

**Comment by gorgias0** [comment]
Hi,
I have the same problem with poor sound quality.
I tried the above command to see the log from volspotconnect 2 and everything looks ok except for this line:
[Vollibrespot] : Reducing normalisation factor to prevent clipping. Please add                                                                 negative pregain to avoid.
